tests/test_streams.py

```diff
@@ -30,6 +35,38 @@ def protocol():
     return mock.Mock(_reading_paused=False)
 
 
+MEMLEAK_SKIP_TYPES = (
+    *(getattr(types, name) for name in types.__all__ if name.endswith('Type')),
+    mock.Mock,
+    abc.ABCMeta,
+)
+
+
+def get_memory_usage(obj):
+    objs = [obj]
+    # Memory leak may be caused by leaked links to same objects.
+    # Without link counting, [1,2,3] is indistiguishable from [1,2,3,3,3,3,3,3]
+    known = defaultdict(int)
```

Comment:
@socketpair FYI there's `collections.Counter` existing exactly for this use-case :)